api/dbadapters/postgres.ts

```diff
@@ -63,15 +63,33 @@ export class PostgresDbAdapter implements IDbAdapter {
       includeQueryInError?: boolean;
     } = { rowLimit: 1000, byteLimit: 1024 * 1024 }
   ) {
-    try {
-      const rows = await this.queryExecutor.execute(statement, options);
-      return { rows, metadata: {} };
-    } catch (e) {
-      if (options.includeQueryInError) {
-        throw new Error(`Error encountered while running "${statement}": ${e.message}`);
-      }
-      throw new ErrorWithCause(`Error executing postgres query: ${e.message}`, e);
-    }
+    return await this.withClientLock(executor => executor.execute(statement, options));
+  }
+
+  public async withClientLock<T>(callback: (client: IDbClient) => Promise<T>) {
+    return await this.queryExecutor.withClientLock(
+      async client =>
+        await callback({
```

Comment:
Do we really need to have a nested async await here?

Reply:
nice, cleaned up